server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1648,8 +1648,7 @@ public void doInTransactionWithoutResult(TransactionStatus status) {
                 return null;
             }
         } else {
-            s_logger.error("UpdateVmNicIpCmd is not supported in this network...");
-            return null;
+            throw new InvalidParameterValueException("UpdateVmNicIpCmd is not supported in L2 network");
```

Comment:
I'd say it is cleaner to move this to line 1545 as
```suggestion
		if(! Network.GuestType.Isolated.equals(network.getGuestType()) {
            throw new InvalidParameterValueException("UpdateVmNicIpCmd is only supported in GUEST networks");
        }
```
#nitpicking